### PR TITLE
Rewrite barcode reader in go

### LINKED ISSUE #1049: Rewrite barcode reader in go
**Is your feature request related to a problem? Please describe.**
Barcode reader is still written in python, adding more dependencies to our stack.

**Describe the solution you'd like**
Rewrite in Go

### PR DESCRIPTION
**Pull request by Scarjit**
# Description

This pr implements barcodereader in go.

Fixes #1049 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Starts in scan only mode (no kafka)
- [X] Tested various barcodes in scan only mode

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have checked that my issue complies with the [Contributing guidelines](https://github.com/united-manufacturing-hub/united-manufacturing-hub/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by Scarjit** [comment]
Does not contain kubernetes changes as of now

**Comment by Scarjit** [comment]
This pr also moves the old python code into the archive repository: https://github.com/united-manufacturing-hub/umh-archive/commit/d8e4c885bb8df01fc7be9e1d2e118dbb4b813aad

**Comment by Scarjit** [comment]
This is tested against the barcodes in /deployment/barcodereader/testdata (generated by https://barcode.tec-it.com/en/)

File: golang/cmd/barcodereader/main.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+// GetBarcodeReaderDevice returns the first barcode reader device found, by name or device path.
+// If no device is found, it will print all available devices and return false, nil.
+func GetBarcodeReaderDevice() (bool, *evdev.InputDevice) {
+	// This could be /dev/input/event0, /dev/input/event1, etc.
+	devicePath := os.Getenv("INPUT_DEVICE_PATH")
+	// This could be "Datalogic ADC, Inc. Handheld Barcode Scanner"
+	deviceName := os.Getenv("INPUT_DEVICE_NAME")
+	unset := false
+	if devicePath == "" && deviceName == "" {
+		zap.S().Infof("No device path or name specified (INPUT_DEVICE_PATH and INPUT_DEVICE_NAME)")
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/barcodereader/main.go:85]: Won't.
It sets unset to true, which will return false, nil in Line 108, thus setting foundDevice to false in main, resulting in os.Exit(1)

**Review by JeremyTheocharis — APPROVED** [review]

**Pull request merged**